Civi/Sepa/Lock/SepaBatchLock.php

```diff
@@ -58,17 +68,31 @@ public function acquire(?int $timeout = NULL, ?string $asyncLockId = NULL): bool
       return TRUE;
     }
 
-    if ($this->asyncLock->isFree()) {
+    $asyncLockAcquireTime = $this->asyncLock->getAcquireTime();
+    if (NULL === $asyncLockAcquireTime || $this->asyncLock->isAcquired()) {
       return TRUE;
     }
 
-    // There is a small chance that the CiviCRM lock is acquired during a page
-    // switch when doing an async run. Thus, we release the lock wait a short
-    // time to give a possible async run the possibility to acquire the lock
-    // again. If the lock isn't acquired then, there's no async update running,
-    // but the async lock wasn't released for some reason.
+    /*
+     * It is possible that the CiviCRM lock is acquired when during a queue
+     * execution an HTTP request was finished and the next one is not running,
+     * yet. Thus, we release the lock and wait some time to give a queue item
+     * the possibility to acquire the lock again. If the async lock time isn't
+     * changed, but the async lock wasn't released for some reason.
+     */
     $this->civiLock->release();
-    sleep(3);
+    for ($i = 0; $i < 5; ++$i) {
```

Comment:
Had me look up again whether pre/post-incrementing in `for` loops has any effect … :nerd_face:

Reply:
As it doesn't I can change it to `$i++` if that increases the readability...